CMakeLists.txt

```diff
@@ -996,21 +997,29 @@ else()
       NAMES
         ocl_icd.h
     )
-    find_library(OCL_ICD_LIBRARIES
-      NAMES
-        OpenCL
-    )
-    if(OCL_ICD_INCLUDE_DIRS AND OCL_ICD_LIBRARIES)
-      set(OCL_ICD_FOUND 1)
-    endif()
+  elseif("${OCL_ICD_LIBRARIES}" STREQUAL "")
+    # OCL_ICD_LIBRARIES is set to empty because of a bug in ocl-icd.pc
+    unset(OCL_ICD_LIBRARIES CACHE)
+  endif()
+
+  # Find OCL_ICD_LIBRARIES with the hint from pkg-config
+  find_library(OCL_ICD_LIBRARIES
```

Comment:
IIUC how `find_library` works, this would find any library named OpenCL in any paths it searches, and that includes "system" paths. But in this case we want to find *only* ocl-icd, not any other OpenCL ICD. So i think this one needs a NO_DEFAULT_PATH

Reply:
Yeah, but that code was also incorrectly leaving OCL_ICD_LIBRARIES set when it found other than ocl-icd loader. But this is just a minor nitpick.